### ISSUE TITLE
List keyboard navigation broken in VS Code 1.31.0

### ISSUE THREAD

**Issue by davewasmer** [VSCodeVim/Vim#3474]
**Describe the bug**
The new list navigation support (#3432 ?) appears to be broken for me. I'm trying to narrow down exactly what the behavior is, and it seems to be timing related.

When I first focus into the file explorer tree (whether via mouse click or a keyboard shortcut), pressing `j` does _not_ navigate down one list item, but instead jumps to the first file that starts with the letter "j".

If immediately hit `j` again, it behaves as expected: the focus moves down one item in the list.

However, if I wait and allow a couple seconds to elapse between the keypresses, it reverts to the broken behavior: it jumps back to the first file starting with `j`. This is true of `k` as well.

I've also noticed that `l` has a similar pattern: subsequent immediate keypresses behave as expected (toggling a directory or opening the file selected). But the first keypress after a short delay jumps to the first list item that begins with "l" (regardless of where it appears - it will jump _up_ directory levels to find it).

**To Reproduce**

1.  Focus into file explorer
2.  Type `j`

Focus jumps to first list item beginning with "j".

**Expected behavior**
Focus moves down one item in the list

**Screenshots**
Focusing into the file explorer and hitting `j`:

![screencast 2019-02-07 12-37-39](https://user-images.githubusercontent.com/707213/52438055-a10e9980-2ad5-11e9-8772-1d141b1b89ff.gif)

**Environment (please complete the following information):**

- Extension (VsCodeVim) version: 1.0.8
- VSCode version: 1.31.0
- OS: macOS Mojave

All other extensions disabled. No custom single letter keyboard shortcuts bound in the file explorer context. `workbench.list.keyboardNavigation` set to `simple`

**Comment by zaynv** [comment]
Yep, this is really annoying as a VIM user who is used to using `j` or `k` to move up/down. Is there any way to disable this new filtering feature for now until there is a way to handle this?

**Comment by jon301** [comment]
The same behavior happens also on previous versions of VsCodeVim too :(

**Comment by davewasmer** [comment]
@zaynv I couldn't find a simple workaround, so I've simply downgraded my VS Code version for now. You can download the previous release [here](https://code.visualstudio.com/updates/v1_30)

**Comment by micimize** [comment]
@jon301 the behavior is due to a new feature in vscode itself, so this is expected - you've got to downgrade.

**Comment by jpoon** [comment]
Dupe of https://github.com/VSCodeVim/Vim/issues/3466

**Comment by davewasmer** [comment]
@jpoon perhaps I'm misreading, but I don't think this is a dup, and I don't see how the linked comment provides resolution.

The comment seems to indicate that the default behavior of the list view should be to respond to traditional vim movement keys (`hjkl`) and only enter filtering mode with `/` is typed.

But as you can see in the gif above in my original post, the default behavior of the list view seems to be neither vim movement or the new filtering functionality in VS Code 1.31. Instead it jumps down the list to the closest matching list item. Plus there's the timing aspect - immediately repeated `j` keystrokes for example revert to vim movement, but delayed repeated keystrokes continue with the jumping behavior.

Would you mind expanding a bit on that linked comment if I've misunderstood it?

**Comment by micimize** [comment]
@jpoon The linked issue is the converse. Their issue is "vim keybindings interfere with filtering", and ours is "filtering interferes with vim keybindings"

**Comment by micimize** [comment]
The [fix](https://github.com/VSCodeVim/Vim/issues/3466#issuecomment-465498306) for #3466 seems to have also fixed this on vscode 1.31 - guess it was a "root cause dupe" after all.

It also seemed to break on vscode 1.30 for me, but the point is I'm up to date and my vim hotkeys are back 🎉 🎉 🎉

**Comment by micimize** [comment]
@jpoon This issue returns after extended usage with Vim `v1.1.0` and vscode `1.31.1`, breaks the plugin entirely on vscode v1.30

![vscode vim keybinding issue](https://user-images.githubusercontent.com/8343799/53454725-bcdad080-39ed-11e9-8232-efaba1346205.gif)

**Comment by davewasmer** [comment]
@jpoon would you consider reopening this, or expanding on the linked comment to help explain how to revert to the old navigation behavior?

**Comment by nosajio** [comment]
I don't use the vim plugin, but I do use "a" and "A" for adding files and dirs while focussed on the tree view. 

I like the filter feature but it seems very self important in its current form. It just takes over all other workbench shortcuts!

Is there a way bind filtering to a shortcut? That way it can be toggled whenever the feature is needed, and shortcuts will still work as expected for affected users.

**Comment by nosajio** [comment]
@micimize no need to downgrade, just change add this to your settings.json:

`"workbench.list.keyboardNavigation": "simple"`

**Comment by martinjaime** [comment]
@nosajio, that does not work for me. The issue described originally persists. Anyone else having this issue after taking @nosajio's suggestion?

**Comment by tommcgurl** [comment]
@martinjaime Ya when I add "workbench.list.keyboardNavigation": "simple" it simple stops working all together. I can't move the cursor and I can't filter to search. 

What worked for me was adding ` "workbench.list.automaticKeyboardNavigation": false` to my `settings.json` 👍

**Comment by brennongs** [comment]
@tommcgurl That got it for me. Thanks! I've been dealing with this for weeks!

**Comment by tommcgurl** [comment]
Awesome @brennongs glad to hear it 👍

**Comment by 020AlievIB** [comment]
Not work for me:
  "workbench.list.keyboardNavigation": "simple",
  "workbench.list.automaticKeyboardNavigation": true,

Vim v1.32.4

Version: 3.2.16 (Universal)
VSCode Version: 1.105.1
